migration-guides/0.14-Migration-Guide.md

```diff
@@ -0,0 +1,597 @@
+
+```purs
+data Proxy :: forall k. k -> Type
+data Proxy a = Proxy
+```
+
+Classes can have signatures too, and they must end with a `Constraint` kind instead of `Type`. For example, here's the new definition of `Prim.Row.Cons`:
+
+```purs
+class Cons :: forall k. Symbol -> k -> Row k -> Row k -> Constraint
+class Cons label a tail row | label a tail -> row, label row -> a tail
+```
+
+### Coercible & Role Annotations
+
+PureScript 0.14 introduces a new compiler-solved class, `Prim.Coerce.Coercible`, to relate types with the same runtime representation. 
+
+You can now use `Safe.Coerce.coerce` from the new `safe-coerce` library in place of `Unsafe.Coerce.unsafeCoerce` to turn an `a` into a `b` when `Coercible a b` holds. If you are currently using `unsafeCoerce` to transform two types with the same runtime representation, you should consider replacing it with `coerce`.
+
+The `Coercible` implementation introduces role annotations for type parameters. Most users do not need to use roles explicitly, as role inference covers the vast majority of cases and this is a new feature that largely does not affect existing code. However, if you would like to understand more about role annotations, please [see the roles documentation in this repository](https://github.com/purescript/documentation/blob/master/language/Roles.md).
+
+### Other Changes
+
+Primes are no longer allowed in JavaScript FFI definitions because PureScript 0.15 will generate ES modules.
+
+**To fix:**
+
+Rename any functions that contain a prime and are implemented via the FFI:
+
+```js
+// before
+exports["functionName'"] = function (a) { return a; }
+
+// after
+exports.functionNameImpl = function (a) { return a; }
+```
+
+You will also need to update the foreign import in your PureScript code:
+
+```diff
+- foreign import functionName' :: forall a. a -> a
++ foreign import functionNameImpl :: forall a. a -> a
++
++ functionName' :: forall a. a -> a
++ functionName' = functionNameImpl
+```
+
+## Library Changes
+
+Several core library changes accompanied the PureScript 0.14 compiler release. This guide covers notable changes to the `purescript` (core), `purescript-contrib`, `purescript-node`, and `purescript-web` organizations.
+
+However, the changes listed below are not exhaustive. As of the PureScript 0.14 release, all libraries in these organizations contain a CHANGELOG.md file which lists any breaking changes in their latest release. You should consult the changelogs for any libraries you depend on in addition to this migration guide.
+
+### Library Restructuring
+
+Several core libraries have been restructured. In almost all cases functionality has been preserved, but some functions, types, and / or instances have been moved to new locations.
```

Comment:
This is not very useful in its current form. We should either say what's actually moved from where to where, or we should just omit this section entirely. I'd prefer the former.

Reply:
Oh never mind, I didn't spot that the following headings are subheadings of this one.